ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt

```diff
@@ -106,15 +107,11 @@ constructor(
    */
   suspend fun loadAndSyncSurveyWithRemote(id: String): Survey? {
     Timber.d("Loading survey $id")
-    val survey =
-      withTimeout(LOAD_REMOTE_SURVEY_TIMEOUT_MILLS) { remoteDataStore.loadSurvey(id) }
-        ?: return null
-
-    localSurveyStore.insertOrUpdateSurvey(survey)
-    survey.tileSources.forEach {
-      tileSourceDao.insertOrUpdateSuspend(it.toLocalDataStoreObject(survey.id))
-    }
-    return survey
+    return withTimeoutOrNull(LOAD_REMOTE_SURVEY_TIMEOUT_MILLS) { remoteDataStore.loadSurvey(id) }
```

Comment:
That's a good question. We might want to consider showing a "retry" button in case something times out. In that scenario, we shouldn't return null and handle the timeout exception at the caller place. Wdyt?

Reply:
This is a "sync" operation. So if it times out, we just abort the task and don't retry.

Reply:
I see. May be worth clarifying w/a comment

Reply:
done